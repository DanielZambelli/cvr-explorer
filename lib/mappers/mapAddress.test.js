const mapAddress = require('./mapAddress')

describe(mapAddress, () => {

  it('[street][nr], [floor]. [door].', () => {
    const result = mapAddress({
      husnummerFra: 2,
      etage: '2',
      sidedoer: 'th',
      conavn: 'Morten Juul Therkildsen',
      vejnavn: 'Anker Heegaards Gade',
      postnummer: 1572,
      postdistrikt: 'København V',
    })
    expect(result.coAddress).toEqual('c/o Morten Juul Therkildsen')
    expect(result.street).toEqual('Anker Heegaards Gade 2, 2. th.')
    expect(result.city).toEqual('København V')
    expect(result.postal).toEqual(1572)
  })

  it('[street][nr][letter], [floor]. [door].', () => {
    const result = mapAddress({
      husnummerFra: 2,
      bogstavFra: 'A',
      etage: '2',
      sidedoer: 'th',
      conavn: 'Morten Juul Therkildsen',
      vejnavn: 'Anker Heegaards Gade',
      postnummer: 1572,
      postdistrikt: 'København V',
    })
    expect(result.street).toEqual('Anker Heegaards Gade 2A, 2. th.')
  })

  it('[street][nr][letter]-[letter], [floor]. [door].', () => {
    const result = mapAddress({
      husnummerFra: 2,
      bogstavFra: 'A',
      bogstavTil: 'B',
      etage: '2',
      sidedoer: 'th',
      conavn: 'Morten Juul Therkildsen',
      vejnavn: 'Anker Heegaards Gade',
      postnummer: 1572,
      postdistrikt: 'København V',
    })
    expect(result.street).toEqual('Anker Heegaards Gade 2A-B, 2. th.')
  })

  it('[street][nr]-[nr], [floor]. [door].', () => {
    const result = mapAddress({
      husnummerFra: 2,
      husnummerTil: 5,
      etage: '2',
      sidedoer: 'th',
      conavn: 'Morten Juul Therkildsen',
      vejnavn: 'Anker Heegaards Gade',
      postnummer: 1572,
      postdistrikt: 'København V',
    })
    expect(result.street).toEqual('Anker Heegaards Gade 2-5, 2. th.')
  })

  it('returns expected keys', () => {
    const result = mapAddress({
      husnummerFra: 2,
      etage: '2',
      sidedoer: 'th',
      conavn: 'Morten Juul Therkildsen',
      vejnavn: 'Anker Heegaards Gade',
      postnummer: 1572,
      postdistrikt: 'København V',
    })
    expect(Object.keys(result)).toMatchSnapshot()
  })

  it('returns expected keys also when nothing is passed', () => {
    const resultFromEmptyObject = mapAddress({})
    const resultFromEmptyArgs = mapAddress()
    expect(Object.keys(resultFromEmptyObject)).toMatchSnapshot()
    expect(Object.keys(resultFromEmptyArgs)).toMatchSnapshot()
  })

})
