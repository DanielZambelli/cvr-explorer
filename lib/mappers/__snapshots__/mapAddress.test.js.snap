// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapAddress returns expected keys 1`] = `
Array [
  "street",
  "city",
  "postal",
  "coAddress",
]
`;

exports[`mapAddress returns expected keys also when nothing is passed 1`] = `
Array [
  "street",
  "city",
  "postal",
  "coAddress",
]
`;

exports[`mapAddress returns expected keys also when nothing is passed 2`] = `
Array [
  "street",
  "city",
  "postal",
  "coAddress",
]
`;
